malai-core/org.malai.ts/src/interaction/TSInteraction.ts

```diff
@@ -61,6 +63,12 @@ export abstract class TSInteraction<D extends InteractionData, F extends FSM<Eve
             eventsToRemove.forEach(type => this.unregisterEventToNode(type, n));
             eventsToAdd.forEach(type => this.registerEventToNode(type, n));
         });
+        if (newState !== this.fsm.initState) {
+            this._registeredTargetNode.forEach(n => {
+                eventsToRemove.forEach(type => this.unregisterEventToNode(type, n));
+                eventsToAdd.forEach(type => this.registerEventToNode(type, n));
+            });
```

Comment:
are you sure that when newState === initState the events registered on the 'to' widgets are unregistered?
I think the code should be 
```typescript
       this._registeredTargetNode.forEach(n => {
              eventsToRemove.forEach(type => this.unregisterEventToNode(type, n));
         });

      if(newState !== this.fsm.initState) {
           this._registeredTargetNode.forEach(n => {
               eventsToAdd.forEach(type => this.registerEventToNode(type, n));
         });
     }
```